app/context/authContext.tsx

```diff
@@ -0,0 +1,204 @@
+'use client';
+
+import React, { createContext, useContext, ReactNode, useMemo, useEffect } from 'react';
+import { useRouter } from 'next/navigation';
+import { MutationStatus, QueryStatus, useMutation, useQuery } from '@tanstack/react-query';
+import { clearSession, getSession, setSession } from '../../utils/actions/session';
+import getAuthUrl from '../../utils/getAuthUrl';
+
+const baseUrl = process.env.NEXT_PUBLIC_API_BASE_URL || '';
+
+export type Role = 'admin' | 'content_manager' | 'member';
+
+export type MiddlewareType = 'auth' | 'guest';
```

Comment:
lets make this enums, we will have  auto-completion with enums and will reduce the risk of making mistakes.